src/lib/logging_events/yojson_prims.ml

```diff
@@ -0,0 +1,47 @@
+(* yojson_prims.ml *)
+
+open Core_kernel
+open Yojson.Safe
+
+(* yojson conversions for (some) built-in OCaml types *)
```

Comment:
There are functions in `Yojson.Safe.Util`, but they return options, not Results. So we'd have to write wrappers around them to get the desired types.

If you use `deriving yojson`, it generates code similar to what's here, rather than calling those functions.

Reply:
We can call functions in `Ppx_deriving_yojson` directly, so it would be best to delegate all of the JSON conversion logic to there.